RFC/src/RFC-0173_Versioning.md

```diff
@@ -0,0 +1,122 @@
+of these versions will be compatible and others not. For example, if a crucial consensus change is added during a hard 
+fork event. Furthermore, there will be multiple networks running Tari code, i.e. Mainnet vs Testnet. Versioning refers 
+to the strategies we will use for nodes to determine if they can communicate.
+
+Tari will contain three different versioning schemes:
+1. WireMode is the first byte a peer sends when connecting to another peer, used to identify the network and/or protocol bytes that follow 
+2. P2P message versions that will accompany every P2P message,
+3. Consensus rules versions that will be exchanged on connection and are included in each block header.
+
+### WireMode byte
+In the Bitcoin P2P protocol messages are preceded by 4 
+[magic values](https://en.bitcoin.it/wiki/Protocol_documentation#Common_structures) or bytes. These values are used to 
+delimit when a new message starts in a byte stream and also are used to indicate which of the Bitcoin networks the node 
+is speaking on, such as TestNet or Mainnet.
+
+Tari message packets are encapsulated using the Noise protocol so we do not need the delimiting functionality of these 
+bytes but Tari will include a single WireMode byte at the beginning of every connection session. This byte will indicate 
+which network a node is communicating on, so that if the counterparty is on a different network it can reject this 
+connection cheaply without having to perform any further operations, like completing the Noise protocol handshake.
+
+### P2P message version
+Peer to Peer messages on the Tari network are encapsulated into message envelopes. The body of message envelopes are 
+defined, serialized and deserialized using Protobuf. These messages will only be updated by adding new fields to the 
+Protobuf definitions, never removing fields. This is done in order to preserve backwards compatibility where newer nodes 
+can still communicate with older nodes. 
+
+The P2P messaging protocol will see many changes in its lifetime. Some will be minor changes that are fully backwards 
+compatible and some changes will be breaking where older nodes will not be able to communicate with newer nodes. In 
+order to document these two classes of changes each P2P message header will contain a `version` field that will use
+a two-part semantic versioning scheme with the format of `major.minor` integer versions. The `minor` version will be 
+incremented whenever there is any change. The `major` version be incremented when there is a breaking change made to 
+the P2P protocol.
+
+### Consensus version
+The final aspect of the Tari system that will be versioned are the Consensus rules. These rules will change as the 
+network matures. Changes to consensus rules can be achieved using either a Soft fork or Hard fork. Soft forks are where 
+new consensus rules are added that older nodes will see as valid (thus backwards compatible) but newer nodes will reject 
```

Comment:
Sure, but the implication is that if they are not aware of the new rules that when they break them the blocks will be rejected. However old nodes will never reject blocks from new nodes because the fork is backwards compatible.

Reply:
Ok fair enough, perhaps I would say (up to you if you want to change it - otherwise just resolve), in the second part of the sentence, that newer nodes may perform new validations and reject incoming blocks.